index.js

```diff
@@ -25,11 +36,19 @@ module.exports = promisedHandlebars
  *   the promised results. This placeholder may not occur in the template or any partial. Neither
  *   my any helper generate the placeholder in to the result. Errors or wrong replacement will
  *   happen otherwise.
+ * @param {Promise} options.Promise the promise implementation to use. Defaults to global.Promise
  * @returns {Handlebars} a modified Handlebars object
  */
 function promisedHandlebars (Handlebars, options) {
   options = options || {}
   options.placeholder = options.placeholder || '\u0001'
+  if (options.Promise) {
```

Comment:
The issue is that the lib won't use `global.Promise`, which sometimes gets configured by the user in a way that won't make through to here when just requiring whatever lib is being used. I suppose you could do something like this:

```js
var promisedHandlebars = require('../')
var Handlebars = promisedHandlebars(require('handlebars'), {Promise: global.Promise})
```

But it doesn't really seem right to go around needing to actively pass in a globals... it kind of defeats the concept of having the global and goes against common intuition about globals.

If we need to maintain backward compatibility and default to `Q` instead of `global.Promise` I think the better solution would be to allow a third option property as `useGlobal`, like this:
```js
var promisedHandlebars = require('../')
var Handlebars = promisedHandlebars(require('handlebars'), {useGlobal: true})
```

An alternative to this would be to add a `./global.js` to the package that does the following,
allowing consumers to use `require('promised-handlebars/global')`.
```js
module.exports = globalPromisedHandlebars
function globalPromisedHandlebars (Handlebars, options) {
  options.Promise = options.Promise || global.Promise
  return require('./')(Handlebars, options)
}
```

This latter option could potentially provide a solution to maintainability going forward since you just keep a branch for 1.x development where the `package.json` has `"main": "q.js"` and you have a `./q.js` file that defaults to q. If we go that route, it may be best to move the main script from `./index.js` to `./lib/index.js` for the sake of clarity in our `require` statements, e.g. in our `./q.js` script:
```js
module.exports = globalPromisedHandlebars
function globalPromisedHandlebars (Handlebars, options) {
  options.Promise = options.Promise || require('q').Promise
  return require('./')(Handlebars, options) // it's not obvious if we get the `main` file here or `/index.js`
  /* vs */
  return require('./lib')(Handlebars, options)
}
```